examples/prom/nonlinear_elasticity_global_rom.cpp

```diff
@@ -1691,16 +1912,812 @@ void RomOperator::Mult_FullOrder(const Vector& vx, Vector& dvx_dt) const
+                    // Compute the row of G corresponding to element e, store in r
+                    ComputeElementRowOfG(ir0, vdofs, ve_j, model, elfun, fe, *eltrans, r);
+
+                    for (int m = 0; m < nqe; ++m)
+                        Gt((e * nqe) + m, j + (i * NB)) = r[m];
+                }
+            }
+
+            if (precondition)
+            {
+                MFEM_ABORT("TODO: Implement preconditioned NNLS for this example");
+            }
+        } // Loop (i) over snapshots
+
+        // Rescale every Gt column (NNLS equation) by its max absolute value.
+        Gt.rescale_cols_max();
+
+        Array<double> const &w_el = ir0->GetWeights();
+        MFEM_VERIFY(w_el.Size() == nqe, "");
+
+        CAROM::Vector w(ne * nqe, true);
+
+        for (int i = 0; i < ne; ++i)
+        {
+            for (int j = 0; j < nqe; ++j)
+                w((i * nqe) + j) = w_el[j];
+        }
+
+        SolveNNLS(rank, nnls_tol, maxNNLSnnz, w, Gt, sol);
+        if (window_ids)
+        {
+            save_CAROM_Vector(sol, "sol_" + std::to_string(oi) + ".csv");
+        }
+    }
+}
+
+// Utility functions
+void WriteMeshEQP(ParMesh *pmesh, const int myid, const int nqe,
+                  CAROM::Vector const &eqpSol)
+{
+    // Find the elements with quadrature points in eqpSol.
+    std::set<int> elements;
+    Vector elemCount(pmesh->GetNE());
+    elemCount = 0.0;
+
+    for (int i = 0; i < eqpSol.dim(); ++i)
+    {
+        if (eqpSol(i) > 1.0e-12)
+        {
+            const int e = i / nqe; // Element index
+            elements.insert(e);
+            elemCount[e] += 1.0;
+        }
+    }
+
+    // Empty sets, since EQP on samples inside elements.
+    std::set<int> faces, edges, vertices;
+    CAROM::SampleVisualization(pmesh, elements, elements, faces, edges,
+                               vertices, "EQPvis", &elemCount);
+}
+
+// Function to compute the indices at which to switch time windows
+void get_window_ids(int n_step, int n_window, CAROM::Vector *ids)
+{
+    int window_size = std::round(n_step / n_window);
+
+    int res = n_step - (n_window * (window_size - 1) + 1);
+    int res_up = res - n_window;
+    int ctr = -1;
+
+    for (int i = 1; i <= n_window + 1; ++i)
+    {
+        ids->item(i - 1) = (i - 1) * window_size + 1 - (i - 1);
+        if (res_up > 0)
+        {
+            if (i <= res - res_up)
+            {
+                ctr += 1;
+            }
+            if (i > 2 && i <= res_up + 1)
+            {
+                ctr += 1;
+            }
+        }
+        else
+        {
+            if (i <= res + 1)
+            {
+                ctr += 1;
+            }
+        }
+        ids->item(i - 1) += ctr;
+    }
+    ids->item(n_window) = n_step + 1;
+}
+
+bool fileExists(const std::string &filename)
+{
+    std::ifstream file(filename);
+    return file.good();
+}
+
+void save_CAROM_Vector(const CAROM::Vector &vector, const std::string &filename)
```

Comment:
`Vector.write` and `Vector.read` are preferable.

Reply:
I wasn't aware of those. I'll change that